src/stepfunctions/steps/sagemaker.py

```diff
@@ -103,6 +116,16 @@ def __init__(self, state_id, estimator, job_name, data=None, hyperparameters=Non
         if isinstance(job_name, (ExecutionInput, StepInput)):
             parameters['TrainingJobName'] = job_name
 
+        if output_path is not None:
+            if isinstance(output_path, (ExecutionInput, StepInput)) or isinstance(output_path, str):
+                parameters['OutputDataConfig']['S3OutputPath'] = output_path
+
+        if data is not None and data_placeholder:
+            # Replace the 'S3Uri' key with one that supports JSONpath value.
+            # Support for uri str only: The list will only contain 1 element
+            temp_data_uri = parameters['InputDataConfig'][0]['DataSource']['S3DataSource'].pop('S3Uri', None)
```

Comment:
Is [`sagemaker.workflow.airflow.training_config`](https://sagemaker.readthedocs.io/en/stable/workflows/airflow/sagemaker.workflow.airflow.html#training-config) going to have all these properties? Seems fragile to do this without any checks.

Reply:
Yes, with the current Sagemaker version we are using, these properties are added whenever with call the function with a str type for _inputs_. Agreed, will add checks to make it more robust and in case Sagemaker makes changes for the format.

Reply:
nit: Can just call it `data_uri`. I don't think there's anything "temporary" about it - there's no existing `data_uri` variable and it's not modified either

Reply:
Agreed! `data_uri` better describes the variable